platform/v1alpha1/types.go

```diff
@@ -0,0 +1,70 @@
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// PlatformOperatorSpec defines the desired state of PlatformOperator.
+type PlatformOperatorSpec struct {
+	// PackageName contains the desired OLM-based Operator package name
+	// that is defined in an existing CatalogSource resource in the cluster.
+	//
+	// This configured package will be managed with the cluster's lifecycle.
+	// +kubebuilder:validation:Required
+	PackageName string `json:"packageName"`
+}
+
+// ActiveBundleDeployment references a BundleDeployment resource.
+type ActiveBundleDeployment struct {
+	// Name is the metadata.name of the referenced BundleDeployment object.
+	// +kubebuilder:validation:Required
+	Name string `json:"name"`
+}
+
+// PlatformOperatorStatus defines the observed state of PlatformOperator
+type PlatformOperatorStatus struct {
+	// Conditions represent the latest available observations of a platform operator's current state.
+	// +optional
+	// +patchMergeKey=type
+	// +patchStrategy=merge
+	// +listType=map
+	// +listMapKey=type
+	Conditions []metav1.Condition `json:"conditions,omitempty"`
+	// ActiveBundleDeployment is the reference to the BundleDeployment resource that's
+	// being managed by this PO resource. If this field is not updated in the status
+	// then it means the PlatformOperator has either not been installed yet or is
+	// failing to install.
```

Comment:
Failures will be reported by conditions right?

Reply:
Yep, that's the intent.